Error: Cannot find module './lib/extend'

From: https://github.com/microsoft/vscode/runs/1719967310#step:10:1

```
Run yarn npm-run-all --max_old_space_size=4095 -lp compile "electron x64" playwright-install download-builtin-extensions
yarn run v1.22.10
$ D:\a\vscode\vscode\node_modules\.bin\npm-run-all --max_old_space_size=4095 -lp compile "electron x64" playwright-install download-builtin-extensions
[compile                    ] $ node --max_old_space_size=4095 ./node_modules/gulp/bin/gulp.js compile
[electron x64               ] $ node build/lib/electron x64
[playwright-install         ] $ node node_modules/playwright/install.js
[download-builtin-extensions] $ node build/lib/builtInExtensions.js
[download-builtin-extensions] [08:00:04] Syncronizing built-in extensions...
[download-builtin-extensions] [08:00:04] You can manage built-in extensions with the --builtin flag
[download-builtin-extensions] internal/modules/cjs/loader.js:818
[download-builtin-extensions]   throw err;
[download-builtin-extensions]   ^
[download-builtin-extensions] 
[download-builtin-extensions] Error: Cannot find module './lib/extend'
[download-builtin-extensions] Require stack:
[download-builtin-extensions] - D:\a\vscode\vscode\node_modules\node.extend\index.js
[download-builtin-extensions] - D:\a\vscode\vscode\node_modules\gulp-remote-retry-src\index.js
[download-builtin-extensions] - D:\a\vscode\vscode\build\lib\extensions.js
[download-builtin-extensions] - D:\a\vscode\vscode\build\lib\builtInExtensions.js
[download-builtin-extensions]     at Function.Module._resolveFilename (internal/modules/cjs/loader.js:815:15)
[download-builtin-extensions]     at Function.Module._load (internal/modules/cjs/loader.js:667:27)
[download-builtin-extensions]     at Module.require (internal/modules/cjs/loader.js:887:19)
[download-builtin-extensions]     at require (internal/modules/cjs/helpers.js:74:18)
[download-builtin-extensions]     at Object.<anonymous> (D:\a\vscode\vscode\node_modules\node.extend\index.js:3:18)
[download-builtin-extensions]     at Module._compile (internal/modules/cjs/loader.js:999:30)
[download-builtin-extensions]     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)
[download-builtin-extensions]     at Module.load (internal/modules/cjs/loader.js:863:32)
[download-builtin-extensions]     at Function.Module._load (internal/modules/cjs/loader.js:708:14)
[download-builtin-extensions]     at Module.require (internal/modules/cjs/loader.js:887:19) {
[download-builtin-extensions]   code: 'MODULE_NOT_FOUND',
[download-builtin-extensions]   requireStack: [
[download-builtin-extensions]     'D:\\a\\vscode\\vscode\\node_modules\\node.extend\\index.js',
[download-builtin-extensions]     'D:\\a\\vscode\\vscode\\node_modules\\gulp-remote-retry-src\\index.js',
[download-builtin-extensions]     'D:\\a\\vscode\\vscode\\build\\lib\\extensions.js',
[download-builtin-extensions]     'D:\\a\\vscode\\vscode\\build\\lib\\builtInExtensions.js'
[download-builtin-extensions]   ]
[download-builtin-extensions] }
[download-builtin-extensions] error Command failed with exit code 1.
[download-builtin-extensions] info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
[compile                    ] [08:00:05] Using gulpfile D:\a\vscode\vscode\gulpfile.js
[compile                    ] [08:00:05] Starting 'compile'...
[compile                    ] [08:00:05] Starting monaco-typecheck ...
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -61,6 +61,7 @@ jobs:
       - name: Create node_modules archive
         if: ${{ steps.cacheNodeModules.outputs.cache-hit != 'true' }}
         run: |
+          mkdir -Force .build
           node build/azure-pipelines/common/listNodeModules.js .build/node_modules_list.txt
           mkdir -Force .build/node_modules_cache
           7z.exe a .build/node_modules_cache/cache.7z -mx3 `@.build/node_modules_list.txt

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -32,13 +32,15 @@ jobs:
       - name: Compute node modules cache key
         id: nodeModulesCacheKey
         run: echo "::set-output name=value::$(node build/azure-pipelines/common/computeNodeModulesCacheKey.js)"
-      - name: Cache node modules
+      - name: Cache node_modules archive
         id: cacheNodeModules
         uses: actions/cache@v2
         with:
-          path: "**/node_modules"
-          key: ${{ runner.os }}-cacheNodeModules11-${{ steps.nodeModulesCacheKey.outputs.value }}
-          restore-keys: ${{ runner.os }}-cacheNodeModules11-
+          path: ".build/node_modules_cache"
+          key: "${{ runner.os }}-cacheNodeModulesArchive-${{ steps.nodeModulesCacheKey.outputs.value }}"
+      - name: Extract node_modules archive
+        if: ${{ steps.cacheNodeModules.outputs.cache-hit == 'true' }}
+        run: 7z.exe x .build/node_modules_cache/cache.7z -aos
       - name: Get yarn cache directory path
         id: yarnCacheDirPath
         if: ${{ steps.cacheNodeModules.outputs.cache-hit != 'true' }}
@@ -56,6 +58,12 @@ jobs:
           PLAYWRIGHT_SKIP_BROWSER_DOWNLOAD: 1
           ELECTRON_SKIP_BINARY_DOWNLOAD: 1
         run: yarn --frozen-lockfile --network-timeout 180000
+      - name: Create node_modules archive
+        if: ${{ steps.cacheNodeModules.outputs.cache-hit != 'true' }}
+        run: |
+          node build/azure-pipelines/common/listNodeModules.js .build/node_modules_list.txt
+          mkdir -Force .build/node_modules_cache
+          7z.exe a .build/node_modules_cache/cache.7z -mx3 `@.build/node_modules_list.txt
 
       - name: Compile and Download
         run: yarn npm-run-all --max_old_space_size=4095 -lp compile "electron x64" playwright-install download-builtin-extensions

